docs/source/receive-adapter.md

```diff
@@ -0,0 +1,174 @@
+# Receive Adapter
```

Comment:
How would we feel if this stuff is actually on the real docs repo ?

Reply:
I think all contributor/developer docs should be as close to the code as possible to avoid having them going out of sync. That helps people that want to contribute to Knative Eventing when they look on the code. 

Everything else such as high level design, how to get started. etc. put in docs .

Reply:
Yes that's essentially the motivation for this doc. Ideally this doc should be updated in the same PR as code change (when needed). 

@matzew does that make sense?